go/internal/cmd.go

```diff
@@ -0,0 +1,55 @@
+// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+package internal
+
+// #include "../lib.h"
+import "C"
+
+import (
+	"fmt"
+
+	"github.com/valkey-io/valkey-glide/go/v2/config"
+	"github.com/valkey-io/valkey-glide/go/v2/internal/errors"
+)
+
+type Batch struct {
+	Commands []Cmd
+	IsAtomic bool
+	Errors   []string // errors processing command args, spotted while batch is filled
+}
+
+type Cmd struct {
+	RequestType uint32 // TODO why C.RequestType doesn't work?
+	Args        []string
+	// Response converter
+	Converter func(any) (any, error)
+}
+
+func MakeCmd(requestType uint32, args []string, converter func(any) (any, error)) Cmd {
+	return Cmd{RequestType: requestType, Args: args, Converter: converter}
+}
+
+func (b Batch) Convert(response []any) ([]any, error) {
+	if len(response) != len(b.Commands) {
+		return nil, &errors.RequestError{
+			Msg: fmt.Sprintf("Response misaligned: received %d responses for %d commands", len(response), len(b.Commands)),
+		}
```

Comment:
Request error was removed in a recent Error cleanup since it had become a dumping ground for all non-specific errors and added no value over the standard error.  
```suggestion
		return nil, fmt.Errorf("Response misaligned: received %d responses for %d commands", len(response), len(b.Commands))
```

Reply:
Will resolve all those after merging #4127